src/CMakeLists.txt

```diff
@@ -1268,303 +1042,17 @@ IF(VISIT_FORTRAN)
-            FOREACH(X ${VISIT_PARALLEL_RPATH})
-                SET(PAR_RPATHS "${PAR_RPATHS} ${X}")
-            ENDFOREACH(X)
-            SET_TARGET_PROPERTIES(${target} PROPERTIES
-                INSTALL_RPATH ${PAR_RPATHS}
-            )
-        ENDIF(VISIT_PARALLEL_RPATH)
-      ENDIF(VISIT_PARALLEL_CXXFLAGS)
-    ELSE(UNIX)
-          ADD_TARGET_INCLUDE(${target} ${VISIT_PARALLEL_INCLUDE})
-          ADD_TARGET_DEFINITIONS(${target} ${VISIT_PARALLEL_DEFS})
-          TARGET_LINK_LIBRARIES(${target} ${VISIT_PARALLEL_LIBS})
-    ENDIF(UNIX)
-
-    # If we're on doing this "nolink mpi" option, we rely on the
-    # PARALLEL_TARGET_LINK_LIBRARIES function to actually link the
-    # target with MPI.
-    IF(NOT VISIT_NOLINK_MPI_WITH_LIBRARIES) # This is a new if test
-        TARGET_LINK_LIBRARIES(${target} ${VISIT_EXE_LINKER_FLAGS} ${VISIT_PARALLEL_LIBS})
-    ENDIF(NOT VISIT_NOLINK_MPI_WITH_LIBRARIES)
-ENDFUNCTION(ADD_PARALLEL_EXECUTABLE)
-
-FUNCTION(ADD_PARALLEL_FORTRAN_EXECUTABLE target)
-    ADD_EXECUTABLE(${target} ${ARGN})
-    IF(VISIT_PARALLEL_FORTRAN_FLAGS)
-        SET(PAR_COMPILE_FLAGS "")
-        FOREACH(X ${VISIT_PARALLEL_FORTRAN_FLAGS})
-            SET(PAR_COMPILE_FLAGS "${PAR_COMPILE_FLAGS} ${X}")
-        ENDFOREACH(X)
-        SET_TARGET_PROPERTIES(${target} PROPERTIES
-            COMPILE_FLAGS ${PAR_COMPILE_FLAGS}
-        )
-
-        IF(VISIT_PARALLEL_FORTRAN_LINKER_FLAGS)
-            SET(PAR_LINK_FLAGS "")
-            FOREACH (X ${VISIT_PARALLEL_FORTRAN_LINKER_FLAGS})
-                SET(PAR_LINK_FLAGS "${PAR_LINK_FLAGS} ${X}")
-            ENDFOREACH (X)
-            SET_TARGET_PROPERTIES(${target} PROPERTIES
-                LINK_FLAGS ${PAR_LINK_FLAGS}
-            )
-        IF(VISIT_PARALLEL_RPATH)
-            SET(PAR_RPATHS "")
-            FOREACH(X ${CMAKE_INSTALL_RPATH})
-                LIST(APPEND PAR_RPATHS ${X})
-            ENDFOREACH(X)
-            FOREACH(X ${VISIT_PARALLEL_RPATH})
-                LIST(APPEND PAR_RPATHS ${X})
-            ENDFOREACH(X)
-            SET_TARGET_PROPERTIES(${target} PROPERTIES
-                INSTALL_RPATH ${PAR_RPATHS}
-            )
-        ENDIF(VISIT_PARALLEL_RPATH)
-        ENDIF(VISIT_PARALLEL_FORTRAN_LINKER_FLAGS)
-        TARGET_LINK_LIBRARIES(${target} ${VISIT_EXE_LINKER_FLAGS} ${VISIT_PARALLEL_FORTRAN_LIBS})
-    ENDIF(VISIT_PARALLEL_FORTRAN_FLAGS)
-ENDFUNCTION(ADD_PARALLEL_FORTRAN_EXECUTABLE)
-
-FUNCTION(PARALLEL_EXECUTABLE_LINK_LIBRARIES target)
-    IF(VISIT_NOLINK_MPI_WITH_LIBRARIES)
-        TARGET_LINK_LIBRARIES(${target} ${ARGN} link_mpi_libs)
-    ELSE(VISIT_NOLINK_MPI_WITH_LIBRARIES)
-        IF(VISIT_STATIC)
-            TARGET_LINK_LIBRARIES(${target} ${ARGN} ${VISIT_PARALLEL_LIBS})
-        ELSE(VISIT_STATIC)
-            TARGET_LINK_LIBRARIES(${target} ${ARGN})
-        ENDIF(VISIT_STATIC)
-    ENDIF(VISIT_NOLINK_MPI_WITH_LIBRARIES)
-ENDFUNCTION(PARALLEL_EXECUTABLE_LINK_LIBRARIES)
+    include(${VISIT_SOURCE_DIR}/CMake/VisItParallel.cmake)
```

Comment:
I think this could be useful but feel free to take it or leave it
```suggestion
    include(${VISIT_SOURCE_DIR}/CMake/VisItParallel.cmake)
    message(STATUS "PARALLEL version of VisIt")
```